zerver/tornado/event_queue.py

```diff
@@ -229,6 +230,21 @@ def accepts_event(self, event: Mapping[str, Any]) -> bool:
             # events are sent only if user_settings_object is False,
             # otherwise only 'user_settings' event is sent.
             return False
+
+        if (
+            event["type"] == "update_global_notifications"
+            and event["notification_name"] not in UserProfile.notification_settings_legacy
+        ) or (
+            event["type"] == "update_display_settings"
+            and event["setting_name"] not in UserProfile.display_settings_legacy
+            and event["setting_name"] != "timezone"
+        ):
+            # TODO: Add tests to verify this when we add a new setting.
+
+            # 'update_display_settings' and 'update_global_notifications'
+            # are expected to send events for settings added before
+            # 'user_settings' event is added in Feature level 89.
+            return False
```

Comment:
I think instead of doing this, we can simply add an if condition in `do_set_user_setting`, etc. since it only depends on setting and not on client or anything else.

Reply:
Ah, that's much better. Thanks, changed.

Reply:
Looks good.